### PR TITLE
upgrading yeoman-generator to 1.0.0

### PR DESCRIPTION
**Pull request by Toddses**
There is some deprecated functionality, as well as the preference for the ES6 class syntax. This is just a first pass as the main app generator. We will want to upgrade the various other generators before merging into master.

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 'use strict';
-var yeoman = require('yeoman-generator'),
-	chalk = require('chalk'),
-	yosay = require('yosay'),
-	mkdirp = require('mkdirp');
+var Generator = require('yeoman-generator'),
+		chalk     = require('chalk'),
+		yosay     = require('yosay'),
+		mkdirp    = require('mkdirp'),
+		copydir   = require('copy-dir');
```
Comment (Toddses) [inline generators/app/index.js]: added copy-dir because `generator.directory` has been removed.

File: generators/app/index.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
-module.exports = yeoman.Base.extend({
-	prompting: function () {
+module.exports = class extends Generator {
```
Comment (Toddses) [inline generators/app/index.js:8]: ES6 class syntax is preferred

File: package.json [hunk 3]
```diff
@@ -50,5 +51,5 @@
     "prepublish": "gulp prepublish",
     "test": "gulp"
   },
-  "license": "unlicense"
+  "license": "SEE LICENSE IN LICENSE"
```
Comment (Toddses) [inline package.json:54]: this just gets rid of an error when installing the dependencies

File: generators/factory/index.js [hunk 4]
```diff
@@ -41,7 +43,7 @@ module.exports = yeoman.Base.extend({
         };
 
     this.fs.copyTpl(
-      getTemplate('factory.js'),
+      getTemplate('factory/factory.js'),
```
Comment (Toddses) [inline generators/factory/index.js:46]: fixes a bug with the template path

File: .editorconfig [hunk 5]
```diff
@@ -3,8 +3,14 @@ root = true
...
 trim_trailing_whitespace = true
 insert_final_newline = true
 
 [*.md]
 trim_trailing_whitespace = false
+
+[*.js]
+indent_style = space
+indent_size = 2
+
```
Comment (Toddses) [inline .editorconfig:16]: I found the original to be irritating when it came to formatting. Perhaps I don't understand how the .editorconfig should work, so we should discuss and get on the same page.

File: generators/app/index.js [hunk 6]
```diff
@@ -1,123 +1,125 @@
...
+    mkdirp    = require('mkdirp'),
+    copydir   = require('copy-dir');
 
-module.exports = yeoman.Base.extend({
-	prompting: function () {
-		// Have Yeoman greet the user.
-		this.log(yosay(
-			'Welcome to the ' + chalk.red('TricomB2B') + ' generator!'
-		));
+module.exports = class extends Generator {
```
Comment (Toddses) [inline generators/app/index.js:8]: yeoman-generator preferred ES6 class syntax

File: generators/app/index.js [hunk 7]
```diff
@@ -1,123 +1,125 @@
...
+    this.fs.copy(getTemplate('view/view.js'), this.destinationPath('src/views/Home/Home.js'));
+    this.fs.copy(getTemplate('view/view.scss'), this.destinationPath('src/views/Home/Home.scss'));
 
-		this.fs.copy(getTemplate('.htaccess'), this.destinationPath('.htaccess'));
-		this.fs.copy(getTemplate('editorconfig.txt'), this.destinationPath('.editorconfig'));
-		this.fs.copy(getTemplate('data.json'), this.destinationPath('data.json'));
-		this.fs.copy(getTemplate('gulpfile.js'), this.destinationPath('gulpfile.js'));
-		this.fs.copy(getTemplate('readme.md'), this.destinationPath('readme.md'));
-		this.fs.copy(getTemplate('js/data-factory.js'), this.destinationPath('src/factories/'+properties.prefixedName+'data/'+properties.prefixedName+'data.factory.js'));
+    copydir.sync(getTemplate('scss'), this.destinationPath('src/scss/'));
```
Comment (Toddses) [inline generators/app/index.js:79]: `this.directory()` was removed in generator 1.0.0. `copy-dir` package handles the copy now.

**Comment by Toddses** [comment]
looks like a lot of changes but largely it's file formatting. requesting a review from Devon to look over and discuss editor config.

**Review by Taylord93 — APPROVED** [review]
Looks good.

**Comment by Taylord93** [comment]
Accidentally hit the merge button... reverted changes.

**Comment by Taylord93** [comment]
You may have to re-open this pr.

**Pull request merged**